src/frequenz/sdk/timeseries/ev_charger_pool/_system_bounds_tracker.py

```diff
@@ -0,0 +1,144 @@
+                sum(
+                    data.active_power_inclusion_lower_bound
+                    for data in self._latest_component_data.values()
+                )
+            ),
+            upper=Power.from_watts(
+                sum(
+                    data.active_power_inclusion_upper_bound
+                    for data in self._latest_component_data.values()
+                )
+            ),
+        )
+        exclusion_bounds = Bounds(
+            lower=Power.from_watts(
+                sum(
+                    data.active_power_exclusion_lower_bound
+                    for data in self._latest_component_data.values()
+                )
+            ),
+            upper=Power.from_watts(
+                sum(
+                    data.active_power_exclusion_upper_bound
+                    for data in self._latest_component_data.values()
+                )
+            ),
+        )
+
+        if (
+            self._last_sent_bounds is None
+            or inclusion_bounds != self._last_sent_bounds.inclusion_bounds
+            or exclusion_bounds != self._last_sent_bounds.exclusion_bounds
+        ):
+            self._last_sent_bounds = SystemBounds(
+                timestamp=max(
+                    msg.timestamp for msg in self._latest_component_data.values()
+                ),
+                inclusion_bounds=inclusion_bounds,
+                exclusion_bounds=exclusion_bounds,
+            )
+            await self._bounds_sender.send(self._last_sent_bounds)
+
+    async def _run_forever(self) -> None:
+        """Run the status tracker forever."""
+        while True:
+            try:
+                await self._run()
+            except Exception:  # pylint: disable=broad-except
+                _logger.exception(
+                    "Restarting after exception in EVChargerSystemBoundsTracker.run()"
+                )
+                await asyncio.sleep(1.0)
+
+    async def _run(self) -> None:
+        """Run the system bounds tracker."""
+        api_client = connection_manager.get().api_client
+        status_rx = self._status_receiver
+        ev_data_rx = merge(
+            *(
+                await asyncio.gather(
+                    *[api_client.ev_charger_data(cid) for cid in self._component_ids]
+                )
+            )
+        )
+
+        async for selected in select(status_rx, ev_data_rx):
+            if selected_from(selected, status_rx):
+                self._component_pool_status = selected.message
+                to_pop = []
+                for comp_id in self._latest_component_data:
+                    if (
+                        comp_id not in self._component_pool_status.working
+                        and comp_id not in self._component_pool_status.uncertain
+                    ):
+                        to_pop.append(comp_id)
+                for comp_id in to_pop:
+                    self._latest_component_data.pop(comp_id, None)
```

Comment:
I was a bit confused at first why `pop()` is used here and not `del self._latest_component_data[comp_id]`.
I'd suggest to write `to_remove` or `to_delete` here though, it's a bit clearer. `pop` kind of implies to me that we want to do something with the result.